pkg/driver/node.go

```diff
@@ -89,11 +90,18 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
 		d.inFlight.Delete(req)
 	}()
 
-	source, ok := req.PublishContext[DevicePathKey]
+	devicePath, ok := req.PublishContext[DevicePathKey]
 	if !ok {
 		return nil, status.Error(codes.InvalidArgument, "Device path not provided")
 	}
 
+	source, err := d.findDevicePath(devicePath, volumeID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Device path %s does not exist. %v", devicePath, err)
```

Comment:
This error is misleading, it's not always that it doesn't exist, there could be other errors.